src/eureka/S5_lightcurve_fitting/models/FleckModel.py

```diff
@@ -0,0 +1,245 @@
+            for c in range(self.nchannel_fitted):
+                chan = self.fitted_channels[c]
+                if self.ld_from_S4:
+                    ld_array = self.ld_array[len_params-2]
+                else:
+                    ld_array = self.ld_array
+                for u in self.coeffs:
+                    index = np.where(np.array(self.paramtitles) == u)[0]
+                    if len(index) != 0:
+                        item = self.longparamlist[c][index[0]]
+                        param = int(item.split('_')[0][-1])
+                        ld_val = ld_array[chan][param-1]
+                        log.writelog(f"{item}, {ld_val}")
+                        # Use the file value as the starting guess
+                        self.parameters.dict[item][0] = ld_val
+                        # In a normal prior, center at the file value
+                        if (self.parameters.dict[item][-1] == 'N' and
+                                self.recenter_ld_prior):
+                            self.parameters.dict[item][-3] = ld_val
+                        # Update the non-dictionary form as well
+                        setattr(self.parameters, item,
+                                self.parameters.dict[item])                     
+
+    def eval(self, channel=None, pid=None, **kwargs):
+        """Evaluate the function with the given values.
+
+        Parameters
+        ----------
+        channel : int; optional
+            If not None, only consider one of the channels. Defaults to None.
+        pid : int; optional
+            Planet ID, default is None which combines the eclipse models from
+            all planets.
+        **kwargs : dict
+            Must pass in the time array here if not already set.
+
+        Returns
+        -------
+        lcfinal : ndarray
+            The value of the model at the times self.time.
+        """
+        if channel is None:
+            nchan = self.nchannel_fitted
+            channels = self.fitted_channels
+        else:
+            nchan = 1
+            channels = [channel, ]
+            
+        if pid is None or pid == 0:
+            pid_iter = range(self.num_planets)
+            pid_id = ''
+        else:
+            pid_iter = [pid,]
+            pid_id = f'_pl{pid}'
+
+        # Get the time
+        if self.time is None:
+            self.time = kwargs.get('time')
+
+        # find number of spots
+        spotinds = [s for s in self.parameters.dict.keys() if 'spotrad' in s]   
+        counter = [s for s in spotinds if '_' in s]
+        nspots = len(spotinds) - len(counter)
+        
+        # Get relevant spot parameters
+        spotrad = np.zeros((self.nchannel_fitted, nspots))
+        spotlat = np.zeros((self.nchannel_fitted, nspots))
+        spotlon = np.zeros((self.nchannel_fitted, nspots))
+        spotcon = np.zeros((self.nchannel_fitted))
+        starrot = np.zeros((self.nchannel_fitted))
+        starinc = np.zeros((self.nchannel_fitted))
```

Comment:
See my comments on the starry model about better defaults for some of these parameters